packages/bsky/src/api/app/bsky/purchase/getFeatures.ts

```diff
@@ -0,0 +1,50 @@
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { Features } from '@atproto/api/dist/client/types/app/bsky/purchase/getFeatures'
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.purchase.getFeatures({
+    auth: ctx.authVerifier.standard,
+    handler: async ({ auth }) => {
+      const viewer = auth.credentials.iss
+
+      const features = await getFeaturesForViewerEntitlements(ctx, viewer)
+
+      return {
+        encoding: 'application/json',
+        body: {
+          features,
+        },
+      }
+    },
+  })
+}
+
+const defaultFeatures: Features = {
+  customProfileColor: false,
+}
+
+const coreEntitlementFeatures: Features = {
+  customProfileColor: true,
+}
+
+const getFeaturesForViewerEntitlements = async (
+  ctx: AppContext,
+  viewerDid: string,
+): Promise<Features> => {
+  const { purchaseEntitlements } = await ctx.dataplane.getPurchaseEntitlements({
+    dids: [viewerDid],
+  })
+
+  if (purchaseEntitlements?.length === 0) {
+    return defaultFeatures
+  }
```

Comment:
Not sure if `purchaseEntitlements` can actually be `undefined`, but we might want something like `!purchaseEntitlements || purchaseEntitlements.length === 0` to avoid referencing `undefined[0]`.